src/binary-writer.cc

```diff
@@ -323,8 +323,9 @@ Index BinaryWriter::GetSymbolIndex(RelocType reloc_type, Index index) {
       type = SymbolType::Global;
       name = module_->globals[index]->name;
       break;
+    case RelocType::TypeIndexLEB:
```

Comment:
Maybe just return `index` here with a comment about that fact that TypeIndexLEB doing use symbol indexes at all?  Would make this change smaller I think.

As it stands this change isn't having any effect at all right?  since it will drop through the default case and print the warning just like before.

Reply:
yeah it doesn't have any effect, I just put it there so the comment makes sense. We could return the index, but personally I would prefer `assert`ing that it's not a type index. Or otherwise change the function name to reflect that it doesn't necessarily deal with symbols.
This is not a very strong preference though.

Reply:
I think having it return the right thing here make sense.

If you want to rename the function thats is fine, but I think a comment explaining that the "symbol index" for this type of relocation is not actually a symbol should be OK.   At least I can't think of a more descriptive name for this function. 

The return value of this function is assigned to the symbol index of the relocation so in some sense we are returning something that well be treated like a symbol index.

Reply:
Yeah that makes sense, I'll do that.